src/ranch_acceptors_sup.erl

```diff
@@ -26,28 +26,41 @@ start_link(Ref, NumAcceptors, Transport) ->
 init([Ref, NumAcceptors, Transport]) ->
 	TransOpts = ranch_server:get_transport_options(Ref),
 	Logger = maps:get(logger, TransOpts, error_logger),
-	SocketOpts = maps:get(socket_opts, TransOpts, []),
-	%% We temporarily put the logger in the process dictionary
-	%% so that it can be used from ranch:filter_options. The
-	%% interface as it currently is does not allow passing it
-	%% down otherwise.
-	put(logger, Logger),
-	LSocket = case Transport:listen(SocketOpts) of
-		{ok, Socket} ->
-			erase(logger),
-			Socket;
-		{error, Reason} ->
-			listen_error(Ref, Transport, SocketOpts, Reason, Logger)
-	end,
-	{ok, Addr} = Transport:sockname(LSocket),
+	LSockets = [{_, BaseSocket}|_] = start_listen_sockets(Ref, NumAcceptors, Transport, TransOpts, Logger),
+	{ok, Addr} = Transport:sockname(BaseSocket),
 	ranch_server:set_addr(Ref, Addr),
 	Procs = [
 		{{acceptor, self(), N}, {ranch_acceptor, start_link, [
 			Ref, N, LSocket, Transport, Logger
 		]}, permanent, brutal_kill, worker, []}
-			|| N <- lists:seq(1, NumAcceptors)],
+			|| {N, LSocket} <- LSockets],
 	{ok, {{one_for_one, 1, 5}, Procs}}.
 
+-spec start_listen_sockets(any(), pos_integer(), module(), #{}, module()) -> [{pos_integer(), inet:socket()}].
+start_listen_sockets(Ref, NumAcceptors, Transport, TransOpts, Logger) ->
+	SocketOpts0 = maps:get(socket_opts, TransOpts, []),
+	BaseSocket = start_listen_socket(Ref, Transport, SocketOpts0, Logger),
+	case maps:get(dedicated_acceptor_sockets, TransOpts, false) of
+		true ->
+			{ok, {_, Port}} = Transport:sockname(BaseSocket),
+			SocketOpts1 = [{port, Port}|proplists:delete(port, SocketOpts0)],
```

Comment:
The socket options are not really a proplist, as multiple raw options can be given to it. The `port` option can be specified only once, though, which is why `proplists:delete` works here. Objections/suggestions?

Reply:
changed this to use `lists:keydelete/3` instead